File: szulogyerek.java

	class Szulo {
		public void fuggveny() {
			System.out.println("szulo");
		};
	}

	class Gyerek extends Szulo {

		@Override
		public void fuggveny() {
			System.out.println("gyerek");
		}
		public void ujfuggveny() {
			System.out.println("ujfuggveny");
		}

}

public class szulogyerek {

	public static void main(String[] args) {
		Szulo a = new Szulo();
		Gyerek b = new Gyerek();
		Szulo c = new Gyerek();

		a.fuggveny();
		b.fuggveny();
		c.fuggveny();
	} 
}

File: szulogyerek.cpp
#include <iostream>
using namespace std;


class Szulo{
    public:
        virtual void fuggveny(){
            cout<<"szulo"<<endl;
        };
};

class Gyerek: public Szulo{
    public:
        void fuggveny() {
            cout<<"gyerek"<<endl;
        }
        void ujfuggveny() {
            cout<<"ujfuggveny"<<endl;
        }
};

int main() {
    Szulo *a;
    a = new Szulo();

    Gyerek *b;
    b = new Gyerek();

    Szulo *c;
    c = new Gyerek();

    //Gyerek *ujszulo;

    a->fuggveny();
    b->fuggveny();
    c->fuggveny();
    
    return 0;
}
